pkg/utils/utils.go

```diff
@@ -53,7 +57,7 @@ func GenerateRuntimeObjectName(backstageCRName string, objectType string) string
 
 // GenerateVolumeNameFromCmOrSecret generates volume name for mounting ConfigMap or Secret
 func GenerateVolumeNameFromCmOrSecret(cmOrSecretName string) string {
-	return cmOrSecretName
+	return ToKubernetesResourceName(cmOrSecretName)
```

Comment:
Well it is hardly something we can call "Kubernetes name", what we need here is simplier conversion from RFC 1123 DNS subdomain name to RFC 1123 label. Which means removing/replacing dots and truncating the string to 63 because cmOrSecretName is already valid RFC 1123 DNS subdomain name.
Also, there is unaddressed problem in  case if transformed string will fit other (the same) name. It is very rare case IMO but still.
WDYT?

Reply:
> Well it is hardly something we can call "Kubernetes name"
> what we need here is simplier conversion from RFC 1123 DNS subdomain name to RFC 1123 label. Which means removing/replacing dots and truncating the string to 63 because cmOrSecretName is already valid RFC 1123 DNS subdomain name.
>

As discussed together, I renamed this to `ToRFC1123Label`. See [`aedd2d1` (#363)](https://github.com/janus-idp/operator/pull/363/commits/aedd2d14ad22a901899e9d18c40af252ce890445).

There are some edge cases requiring more than a simple replacement, especially with truncation, where we might end up with invalid names, even if the original cmOrSecretName is valid. Example with a valid CM/Secret name like `ppxkgq-df-yyatvyrgjtwivunibicne-bvyyotvonbrtfv-awylmrez.ksvqjw--zxpgdi`.

> Also, there is unaddressed problem in case if transformed string will fit other (the same) name. It is very rare case IMO but still.

As discussed also, let me know if you find a way to handle this rare case. Otherwise, this can be addressed in a separate issue/PR.